src/ukify/ukify.py

```diff
@@ -219,6 +332,9 @@ def make_uki(opts):
     else:
         linux = opts.linux
 
+    if opts.uname is None:
+        opts.uname = Uname.scrape(opts.linux, opts=opts)
```

Comment:
grmbl, do we really want this? isn#t this better something we expect people to pass in? i mean, during build time this information should be clearly available always?

i mean, this info isn#t strictly necessary, so for people who quickndirty want to build a UKI they can just leave the field out?

Reply:
oh, hmm, kernels are internally ELF, so i guess we could use the package notes infra as primary source of info, no? i mean, it would be a form of dogfooding ;-)

Reply:
I don't like this part too much, it is hacky. And in stuff like package installation scripts it won't be necessary, since you generally know the uname from the package version. But for casual use where people combine an existing kernel with some commandline options and other fluff, I think it's fairly useful. If it turns out that this scraping is too brittle, we can always drop that later.

>  we could use the package notes

I don't think anybody builds the kernel with package notes. On PPC64, the kernel has a note with the version, but afaict, it's not on all architectures and is hard to access anyway.

Reply:
maybe at the very least add a log message "Kernel version was not speicified, resorting to scraping. Yuck. Please specify explicitly." or something like that.

plus an emoji.

Reply:
I used "confounded face": 😖